### PR TITLE
Add the OIDC OAuth setup

### PR DESCRIPTION
**Pull request by jvlcek**
Fixes https://github.com/ManageIQ/manageiq_docs/issues/1443

These documentation changes  address the changes made in [manageiq-appliance/PR 282](https://github.com/ManageIQ/manageiq-appliance/pull/282)

I also moved the `Configuring Keycloak` section to the top, as this order makes a better flow.

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot add_label enhancement
@miq-bot add_label jansa/yes?
@miq-bot assign @abellotti

**Comment by jvlcek** [comment]
@abellotti  Please take a look.

File: auth/openid_connect.adoc [hunk 1]
```diff
@@ -16,13 +16,14 @@ The current implementation only secures the Appliance's Web administrative UI wi
 
 *Note:* The REST API and Self-Service UI do not currently support OIDC.
```
Comment (abellotti) [inline auth/openid_connect.adoc:17]: I guess this Note needs changing since REST API crossed the finish line.

File: auth/openid_connect.adoc [hunk 2]
```diff
@@ -163,6 +113,80 @@ The following _User Session Note_ mappers must be manually created:
...
+OIDCOAuthClientID                | oidc_client_id              | The OpenID-Connect Client ID
+| OIDCClientSecret
+
+end
+
+OIDCOAuthClientSecret            | oidc_client_secret          | The OpenID-Connect Client Secret
+| OIDCOAuthIntrospectionEndpoint | oidc_introspection_endpoint | The OpenID-Connect Introspection Endpoint
+|======================================================================================
+
+Optional parametes can be appended to the end fo the configuration file to avoid SSL verification. These can
```
Comment (abellotti) [inline auth/openid_connect.adoc:172]: parameters

File: auth/openid_connect.adoc [hunk 3]
```diff
@@ -242,14 +266,24 @@ To configure the appliance for OpenID-Connect Authentication:
...
 
+When the `--oidc-introspection-endpoint` is not specified, an attempt will be made to
+fetch the introspection endpoint by querying the OpenID-Connect Provider URL.
+
 When the `--oidc-client-host` is not specified, the configured appliance host
 is used for the ServerName and OpenID-Connect Redirect URI in the
 OpenID-Connect Apache configuration file. The `--oidc-client-host` is
 usually the externally reachable FQDN of the appliance.
 
+When the `--oidc-insecure` is specified not SSL verification will be done when
```
Comment (abellotti) [inline auth/openid_connect.adoc:282]: not -> no

File: auth/openid_connect.adoc [hunk 4]
```diff
@@ -242,14 +266,24 @@ To configure the appliance for OpenID-Connect Authentication:
...
+attempting to fetch the `--oidc-introspection-endpoint`, if it is not specified,
+and SSL verification will be turned off using these parameters as described above
```
Comment (abellotti) [inline auth/openid_connect.adoc:284]: got lost here.

File: auth/openid_connect.adoc (hunk 1 above)
Comment (jvlcek) [reply auth/openid_connect.adoc:17]: @abellotti Great Catch! I hadn't even noticed that! Thank you.

File: auth/openid_connect.adoc (hunk 4 above)
Comment (jvlcek) [reply auth/openid_connect.adoc]: @abellotti I've improved the wording. Thank you.

File: auth/openid_connect.adoc (hunk 3 above)
Comment (jvlcek) [reply auth/openid_connect.adoc]: Know not what not I know. Thank you for untying this knot. Oh no! 😉

**Comment by abellotti** [comment]
This LGTM @jvlcek 👍
Thank you for the OIDC doc update !!

**Pull request merged**